fsutils/romloader/romloader_main.c

```diff
@@ -0,0 +1,103 @@
+ * apps/fsutils/romloader/romloader_main.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/mount.h>
+#include <sys/boardctl.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include <nuttx/drivers/ramdisk.h>
+
+#include "romfs.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define NSECTORS(b)  (((b) + CONFIG_FSUTILS_ROMLOADER_SECTORSIZE - 1) / \
+                      CONFIG_FSUTILS_ROMLOADER_SECTORSIZE)
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * romloader_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  int ret;
+  struct boardioc_romdisk_s desc;
+
+  /* Create a ROM disk for the ROMFS filesystem */
+
+  printf("Registering romdisk at /dev/ram%d\n",
+         CONFIG_FSUTILS_ROMLOADER_DEVMINOR);
+
+  desc.minor    = CONFIG_FSUTILS_ROMLOADER_DEVMINOR;   /* Minor device number of the ROM disk. */
+  desc.nsectors = NSECTORS(romfs_img_len);             /* The number of sectors in the ROM disk */
+  desc.sectsize = CONFIG_FSUTILS_ROMLOADER_SECTORSIZE; /* The size of one sector in bytes */
+  desc.image    = (FAR uint8_t *)romfs_img;            /* File system image */
+
+  ret = boardctl(BOARDIOC_ROMDISK, (uintptr_t)&desc);
+  if (ret < 0)
+    {
+      fprintf(stderr, "ERROR: romdisk_register failed: %s\n",
```

Comment:
Maybe better to use `dprintf(STDERR_FILENO,` instead?

Reply:
Idk, all occurrences in NuttX use ```stderr```. If we use ```STDERR_FILENO```, it would be the only one.

Reply:
From what I saw, ```stderr``` should be used with ```fprintf``` and ```STDERR_FILENO``` with ```dprintf```:

```
#ifdef CONFIG_FILE_STREAM
  fprintf(stderr, "%d: %pV: %s\n", _SCHED_GETTID(), &vaf, strerror(error));
#else
  dprintf(STDERR_FILENO, "%d: %pV: %s\n", _SCHED_GETTID(),
                                          &vaf, strerror(error));
#endif
```

Reply:
I could change it to ```dprintf``` to avoid compilation errors when ```CONFIG_DEFAULT_SMALL``` is enabled.